src/main/resources/public/js/components/catalog-card.js

```diff
@@ -0,0 +1,132 @@
+
+        if (app.target_companies) {
+            buttons.push({
+                className: 'fa fa-briefcase category-icon',
+                text: i18n._(t`store.companies`),
+                classNameButton,
+            });
+        }
+
+        if (app.target_publicbodies) {
+            buttons.push({
+                className: 'fa fa-building category-icon',
+                text: i18n._(t`store.publicbodies`),
+                classNameButton,
+            });
+        }
+
+        return buttons;
+    };
+
+    render = () => {
+        const { app } = this.props;
+        return (
+            <div className="content-wrapper">
+                <div className="cards-container" onClick={() => this.openInstallAppPage()}>
+                    <div id="card" onClick={() => this.openInstallAppPage()}>
+                        <div id="main">
+                            <div id="icon-wrapper">
+                                <img src={app.icon} />
+                            </div>
+                            <div id="content">
+                                <div className="bloc">
+                                    <div id="app-name">
+                                        <div id="name" className="flex-text">
+                                            {app.name}
+                                        </div>
+                                    </div>
+                                </div>
+                                <div id="app-provider">{app.provider}</div>
+                                <div id="app-description">{app.description}</div>
+                            </div>
+                        </div>
+                        <div id="footer">
+                            <div className="footer-infos">
+                                {this.getTags(app).map(tagInfo => (
+                                    <Tag
+                                        className={tagInfo.className}
+                                        text={tagInfo.text}
+                                        classNameButton={tagInfo.classNameButton}
+                                    />
+                                ))}
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        );
+    };
+}
+
+CatalogCard.propTypes = {
+    app: PropTypes.object,
+    config: PropTypes.object,
+};
+
+export default withRouter(CatalogCard);
+
+export class Tag extends React.PureComponent {
+    render() {
+        const { className, text, classNameButton } = this.props;
+
+        return (
+            <button type="button" key="indicator_button" className={`category-btn ${classNameButton}`}>
+                <i className={className}></i>
+                <span className="btn-label">{text}</span>
+            </button>
+        );
+    }
+}
+Tag.propTypes = {
+    className: PropTypes.string,
+    text: PropTypes.string,
+    classNameButton: PropTypes.string,
+};
```

Comment:
Maybe the `<Tag />` component would be happy to have its own file :)

Reply:
Done :)
Better practice indeed.